python.d/monit.chart.py

```diff
@@ -0,0 +1,150 @@
+# update_every = 2
+priority = 60000
+retries = 60
+
+# see enum State_Type from monit.h (https://bitbucket.org/tildeslash/monit/src/master/src/monit.h)
+MONIT_SERVICE_NAMES = [ 'Filesystem', 'Directory', 'File', 'Process', 'Host', 'System', 'Fifo', 'Program', 'Net' ]
+DEFAULT_SERVICES_IDS = [ 0, 1, 2, 3, 4, 6, 7, 8 ]
+
+# charts order (can be overridden if you want less charts, or different order)
+ORDER = [ 'filesystem', 'directory', 'file', 'process', 'process_uptime', 'process_threads', 'process_children', 'host', 'host_latency', 'system', 'fifo', 'program', 'net' ]
+CHARTS = {
+    # id: {
+    #     'options': [name, title, units, family, context, charttype],
+    #     'lines': [
+    #         [unique_dimension_name, name, algorithm, multiplier, divisor]
+    #     ]}
+    'filesystem': {
+        'options': ['filesystems', 'Filesystems', 'filesystems', 'filesystem', 'monit.filesystems', 'line'],
+        'lines': []
+    },
+    'directory': {
+        'options': ['directories', 'Directories', 'directories', 'filesystem', 'monit.directories', 'line'],
+        'lines': []
+    },
+    'file': {
+        'options': ['files', 'Files', 'files', 'filesystem', 'monit.files', 'line'],
+        'lines': []
+    },
+    'fifo': {
+        'options': ['fifos', 'Pipes (fifo)', 'pipes', 'filesystem', 'monit.fifos', 'line'],
+        'lines': []
+    },
+    'program': {
+        'options': ['programs', 'Programs statuses', 'programs', 'applications', 'monit.programs', 'line'],
+        'lines': []
+    },
+    'process': {
+        'options': ['processes', 'Processes statuses', 'processes', 'applications', 'monit.services', 'line'],
+        'lines': []
+    },
+    'process_uptime': {
+        'options': ['processes uptime', 'Processes uptime', 'seconds', 'applications', 'monit.process_uptime', 'line', 'hidden'],
+        'lines': []
+    },
+    'process_threads': {
+        'options': ['processes threads', 'Processes threads', 'threads', 'applications', 'monit.process_threads', 'line'],
+        'lines': []
+    },
+    'process_children': {
+        'options': ['processes childrens', 'Child processes', 'childrens', 'applications', 'monit.process_childrens', 'line'],
+        'lines': []
+    },
+    'host': {
+        'options': ['hosts', 'Hosts', 'hosts', 'network', 'monit.hosts', 'line'],
+        'lines': []
+    },
+    'host_latency': {
+        'options': ['hosts latency', 'Hosts latency', 'milliseconds/s', 'network', 'monit.host_latency', 'line'],
+        'lines': []
+    },
+    'net': {
+        'options': ['interfaces', 'Network interfaces and addresses', 'interfaces', 'network', 'monit.networks', 'line'],
+        'lines': []
+    },
+}
+
+class Service(UrlService):
+    def __init__(self, configuration=None, name=None):
+        UrlService.__init__(self, configuration=configuration, name=name)
+        self.url = self.configuration.get('url', "http://127.0.0.1:2812/_status?format=xml&level=full")
```

Comment:
I assume that you only support parsing the full XML format?  If that's the case, then at minimum the query string (the `?format=xml&level=full` bit) should be hard-coded in the module and not required as part of the URL in the configuration.  I'd say that the `/_status` part probably should be too, as that's part of the monit API, and not something that's likely to be changed by an end user.

Also, please update the IP address to a host name.  The `UrlService` class will handle that correctly with very little overhead, and that way people who are actually running monit on IPv6 over the loopback interface will actually have IPv6 used.

Essentially, something like this, and then update the default value explained in the config file accordingly:

```
    uri = self.configuration.get('uri', 'http://localhost:2812')
    self.uri = format('{0}/_status?format=xml&level=full', uri)
```

Reply:
@n0guest 
replace `url` with `host`, `port` and `scheme`, this whould be better i think
```python
scheme = self.configuration.get('scheme', 'http')
host = self.configuration.get('host', 'localhost')
port = self.configuration.get('port', 2812)
self.url = '{0}://{1}:{2}/_status?format=xml&level=full'.format(scheme, host,  port)
```

Reply:
i believe `/_status` thing in `apache`, `nignx`, `hapoxy` is configurable, then it shouldn't be static.

what about `monit`? does it always `_status?format=xml&level=full` ?

Reply:
Yes, it's only way. To say more, it's not even documented in [official docs](https://mmonit.com/monit/documentation/monit.html#Service-checks), but mentioned many times in [issues on bitbucket](https://bitbucket.org/tildeslash/monit/issues?q=format%3Dxml). So, it's half-official, but only one known way to get stats from Monit.

PS: At stackoverflow there were mentions, that monit also supports JSON (as format=json), but i can't get it working and proceed from parsing XML.

Reply:
Ok, another option is to leave it as is (`url') and add a simple check, like

```python
def check(self):
    if not self.url.endswith('_status?format=xml&level=full'):
        self.error('text')
        return False
```

Reply:
I did as @Ferroin initially suggested (see 6e023743d5154cc0a21a03d537a1ca7b77f03ab3)